src/main/java/it/pagopa/pn/service/desk/service/impl/NotificationAndMessageServiceImpl.java

```diff
@@ -0,0 +1,168 @@
+package it.pagopa.pn.service.desk.service.impl;
+
+import it.pagopa.pn.service.desk.exception.PnGenericException;
+import it.pagopa.pn.service.desk.generated.openapi.msclient.pndelivery.v1.dto.SentNotificationDto;
+import it.pagopa.pn.service.desk.generated.openapi.msclient.pndeliverypush.v1.dto.TimelineElementCategoryV20Dto;
+import it.pagopa.pn.service.desk.generated.openapi.msclient.pndeliverypush.v1.dto.TimelineElementV20Dto;
+import it.pagopa.pn.service.desk.generated.openapi.server.v1.dto.*;
+import it.pagopa.pn.service.desk.mapper.NotificationAndMessageMapper;
+import it.pagopa.pn.service.desk.middleware.externalclient.pnclient.datavault.PnDataVaultClient;
+import it.pagopa.pn.service.desk.middleware.externalclient.pnclient.delivery.PnDeliveryClient;
+import it.pagopa.pn.service.desk.middleware.externalclient.pnclient.deliverypush.PnDeliveryPushClient;
+import it.pagopa.pn.service.desk.service.NotificationAndMessageService;
+import lombok.CustomLog;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import static it.pagopa.pn.service.desk.exception.ExceptionTypeEnum.ERROR_ON_DELIVERY_CLIENT;
+import static it.pagopa.pn.service.desk.exception.ExceptionTypeEnum.ERROR_ON_DELIVERY_PUSH_CLIENT;
+
+@Service
+@CustomLog
+public class NotificationAndMessageServiceImpl implements NotificationAndMessageService {
+
+    private final PnDataVaultClient dataVaultClient;
+    private final PnDeliveryClient pnDeliveryClient;
+    private final PnDeliveryPushClient pnDeliveryPushClient;
+
+    public NotificationAndMessageServiceImpl(PnDataVaultClient dataVaultClient, PnDeliveryClient pnDeliveryClient, PnDeliveryPushClient pnDeliveryPushClient) {
+        this.dataVaultClient = dataVaultClient;
+        this.pnDeliveryClient = pnDeliveryClient;
+        this.pnDeliveryPushClient = pnDeliveryPushClient;
+    }
+
+    @Override
+    public Mono<SearchNotificationsResponse> searchNotificationsFromTaxId(String xPagopaPnUid, OffsetDateTime startDate, OffsetDateTime endDate, Integer size, String nextPagesKey, SearchNotificationsRequest request) {
```

Comment:
questo metodo contiene troppe righe, dividiamolo in più metodi

Reply:
abbiamo diviso in più metodi le righe come richiesto